Sources_v3/StreamChatUI/ChatChannel/ChatChannelMessageComposerView.swift

```diff
@@ -4,169 +4,175 @@
-        button.widthAnchor.constraint(equalTo: button.heightAnchor, multiplier: 1).isActive = true
+    public private(set) lazy var shrinkInputButton: UIButton = {
+        let button = uiConfig.messageComposer.composerButton.init().withoutAutoresizingMaskConstraints
+        button.image = UIImage(named: "shrinkInputArrow", in: .streamChatUI)
         return button
     }()
-
-    public private(set) lazy var boltButton: UIButton = {
-        let button = UIButton().withoutAutoresizingMaskConstraints
-        if #available(iOS 13.0, *) {
-            button.setImage(UIImage(systemName: "bolt.fill"), for: .normal)
-        }
-        button.widthAnchor.constraint(equalTo: button.heightAnchor, multiplier: 1).isActive = true
+    
+    public private(set) lazy var stateIcon: UIImageView = {
+        let imageView = UIImageView().withoutAutoresizingMaskConstraints
+        imageView.contentMode = .center
+        imageView.widthAnchor.constraint(equalTo: imageView.heightAnchor, multiplier: 1).isActive = true
+        return imageView
+    }()
+    
+    public private(set) lazy var dismissButton: UIButton = {
+        let button = uiConfig.messageComposer.composerButton.init().withoutAutoresizingMaskConstraints
+        button.image = UIImage(named: "dismissInCircle", in: .streamChatUI)?
+            .tinted(with: uiConfig.messageComposer.colors.composerButton)
         return button
     }()
     
+    public private(set) lazy var titleLabel: UILabel = {
+        let label = UILabel().withoutAutoresizingMaskConstraints
+        label.textAlignment = .center
+        label.font = .preferredFont(forTextStyle: .headline)
+        label.adjustsFontForContentSizeCategory = true
+        return label
+    }()
+    
     // MARK: - Init
     
-    public required init(
-        uiConfig: UIConfig<ExtraData> = .default
-    ) {
-        self.uiConfig = uiConfig
-        
+    public required init() {
         super.init(frame: .zero, inputViewStyle: .default)
-        
-        commonInit()
     }
     
     public required init?(coder: NSCoder) {
-        uiConfig = .default
-        
         super.init(coder: coder)
+    }
+    
+    // MARK: - Overrides
+    
+    override open func didMoveToSuperview() {
+        super.didMoveToSuperview()
+        guard superview != nil else { return }
         
-        commonInit()
+        setUp()
+        applyDefaultAppearance()
+        setUpAppearance()
+        setUpLayout()
+        updateContent()
     }
     
-    public func commonInit() {
-        embed(container)
-
-        setupAppearance()
-        setupLayout()
+    override open var intrinsicContentSize: CGSize {
+        let size = CGSize(
+            width: UIView.noIntrinsicMetric,
+            height: container.systemLayoutSizeFitting(UIView.layoutFittingCompressedSize).height
+        )
+        return size
     }
     
     // MARK: - Public
     
-    open func setupAppearance() {
+    open func setUp() {
+        uiConfig = .default
```

Comment:
There's no need for doing this. Actually, it breaks the whole system instead :) `UIConfig` is accessible just by conforming to `UIConfigProvider`. It resolves the correct UIConfig instance for you

Reply:
That was leftover from debugging some stuff, thanks.